src/librustc_resolve/lib.rs

```diff
@@ -2934,8 +2934,38 @@ impl<'a> Resolver<'a> {
                                                               here due to private fields"));
                             }
                         } else {
-                            err.span_label(span, format!("did you mean `{} {{ /* fields */ }}`?",
-                                                         path_str));
+                            // HACK(estebank): find a better way to figure out that this was a
```

Comment:
This is .. certainly a hack. I don't know what would be the best way to handle it, though; we have an `id`, so if we had the AST on hand we could find the context for this path by -- at worst -- walking through it. I'm not sure if there is a "map" for the AST that lets you find out (e.g.) the parent node?

That said, this function is invoked (iirc) via a visit walk, so perhaps we could add some kind of "context" parameter and thread the information down.

Reply:
The problem is that the AST is already incorrect. I would like to modify the parser to accept struct literals in these positions to minimize errors being shown and include the suggestion then, but wanted to have a quick solution until then as this is a problem I've seen crop up multiple times on the issue tracker and in the forums.

Reply:
Well, my point regarding hack was not the approach of making a suggestion, which makes sense to me, but rather using the spans etc.

Reply:
OK, re-reading this diff, I see I was a bit confused about what was going on. This now seems more reasonable than I thought at first. =)